services/wallet/router/pathprocessor/processor_nft.go

```diff
@@ -0,0 +1,198 @@
+
+	// Register handlers in order of priority
+	// Specialized handlers first, then generic ERC721
+	processor.handlers = append(processor.handlers, NewCryptoKittiesHandler(rpcClient, transactor))
+	processor.handlers = append(processor.handlers, NewCryptoPunksHandler(rpcClient, transactor))
+	processor.handlers = append(processor.handlers, NewERC721Handler(rpcClient, transactor))
+
+	return processor
+}
+
+func createNFTErrorResponse(err error) error {
+	return createErrorResponse(pathProcessorCommon.ProcessorERC721Name, err)
+}
+
+func (s *NFTProcessor) Name() string {
+	return pathProcessorCommon.ProcessorERC721Name
+}
+
+func (s *NFTProcessor) AvailableFor(params ProcessorInputParams) (bool, error) {
+	if params.FromChain == nil || params.ToChain == nil {
+		return false, ErrNoChainSet
+	}
+	if params.FromToken == nil {
+		return false, ErrNoTokenSet
+	}
+
+	// Only handle same-chain transfers with no destination token (NFT transfers)
+	if params.FromChain.ChainID != params.ToChain.ChainID || params.ToToken != nil {
+		return false, nil
+	}
+
+	return s.getHandlerForContract(params) != nil, nil
+}
+
+func (s *NFTProcessor) CalculateFees(params ProcessorInputParams) (*big.Int, *big.Int, error) {
+	return walletCommon.ZeroBigIntValue(), walletCommon.ZeroBigIntValue(), nil
+}
+
+func (s *NFTProcessor) getHandlerForContractID(contractID thirdparty.ContractID) NFTHandler {
+	for _, handler := range s.handlers {
+		if handler.CanHandle(contractID) {
+			return handler
+		}
+	}
+
+	return nil
+}
+
+func (s *NFTProcessor) getHandlerForContract(params ProcessorInputParams) NFTHandler {
+	contractID := thirdparty.ContractID{
+		ChainID: walletCommon.ChainID(params.FromChain.ChainID),
+		Address: params.FromToken.Address,
+	}
+
+	return s.getHandlerForContractID(contractID)
+}
+
+func (s *NFTProcessor) getHandlerForMultipathTx(sendArgs *MultipathProcessorTxArgs) NFTHandler {
+	contractID := thirdparty.ContractID{
+		ChainID: walletCommon.ChainID(sendArgs.ChainID),
+		Address: common.Address(*sendArgs.ERC721TransferTx.To),
+	}
+
+	return s.getHandlerForContractID(contractID)
+}
+
+func (s *NFTProcessor) PackTxInputData(params ProcessorInputParams) ([]byte, error) {
+	handler := s.getHandlerForContract(params)
+	if handler == nil {
+		return nil, createNFTErrorResponse(ErrNoTokenSet)
+	}
+
+	data, err := handler.PackTxInputData(params)
+	if err != nil {
+		return nil, createNFTErrorResponse(err)
+	}
+
+	return data, nil
+}
+
+func (s *NFTProcessor) EstimateGas(params ProcessorInputParams, input []byte) (uint64, error) {
+	handler := s.getHandlerForContract(params)
+	if handler == nil {
+		return 0, createNFTErrorResponse(ErrNoTokenSet)
+	}
+
+	estimation, err := handler.EstimateGas(params, input, handler.Name())
+	if err != nil {
+		return 0, createNFTErrorResponse(err)
+	}
+	return estimation, nil
+}
+
+func (s *NFTProcessor) Send(sendArgs *MultipathProcessorTxArgs, lastUsedNonce int64, verifiedAccount *generator.Account) (types.Hash, uint64, error) {
```

Comment:
Pretty sure that we can get rid of this and `BuildTransaction` and that way remove `getHandlerForMultipathTx` also, not sure if we use them at all anymore.